exchanges/bybit/bybit.go

```diff
@@ -8,740 +8,2269 @@ import (
+func (by *Bybit) GetIndexPriceKline(ctx context.Context, category, symbol string, interval kline.Interval, startTime, endTime time.Time, limit int64) ([]KlineItem, error) {
+	params, err := fillCategoryAndSymbol(category, symbol)
+	if err != nil {
+		return nil, err
+	}
+	intervalString, err := intervalToString(interval)
+	if err != nil {
+		return nil, err
+	}
+	params.Set("interval", intervalString)
+	if !startTime.IsZero() {
+		params.Set("start", strconv.FormatInt(startTime.UnixMilli(), 10))
+	}
+	if !endTime.IsZero() {
+		params.Set("end", strconv.FormatInt(endTime.UnixMilli(), 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	var resp KlineResponse
+	err = by.SendHTTPRequest(ctx, exchange.RestSpot, common.EncodeURLValues("market/index-price-kline", params), defaultEPL, &resp)
+	if err != nil {
+		return nil, err
+	}
+	return processKlineResponse(resp.List)
 }
 
-func constructOrderbook(o *orderbookResponse) (*Orderbook, error) {
-	var (
-		s   Orderbook
-		err error
-	)
-	s.Bids, err = processOB(o.Data.Bids)
+// GetOrderBook retrieves for orderbook depth data.
+func (by *Bybit) GetOrderBook(ctx context.Context, category, symbol string, limit int64) (*Orderbook, error) {
+	params, err := fillCategoryAndSymbol(category, symbol)
 	if err != nil {
 		return nil, err
 	}
-	s.Asks, err = processOB(o.Data.Asks)
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	var resp orderbookResponse
+	err = by.SendHTTPRequest(ctx, exchange.RestSpot, common.EncodeURLValues("market/orderbook", params), defaultEPL, &resp)
 	if err != nil {
 		return nil, err
 	}
-	s.Time = o.Data.Time.Time()
-	return &s, err
+	return constructOrderbook(&resp)
 }
 
-// GetOrderBook gets orderbook for a given market with a given depth (default depth 100)
-func (by *Bybit) GetOrderBook(ctx context.Context, symbol string, depth int64) (*Orderbook, error) {
-	var o orderbookResponse
-	strDepth := "100" // default depth
-	if depth > 0 && depth < 100 {
-		strDepth = strconv.FormatInt(depth, 10)
+func fillCategoryAndSymbol(category, symbol string, optionalSymbol ...bool) (url.Values, error) {
+	if category == "" {
+		return nil, errCategoryNotSet
+	} else if category != cSpot && category != cLinear && category != cInverse && category != cOption {
+		return nil, fmt.Errorf("%w, category: %s", errInvalidCategory, category)
+	}
+	params := url.Values{}
+	if symbol == "" && (len(optionalSymbol) == 0 || !optionalSymbol[0]) {
+		return nil, errSymbolMissing
+	} else if symbol != "" {
+		params.Set("symbol", symbol)
+	}
+	params.Set("category", category)
+	return params, nil
+}
+
+// GetTickers returns the latest price snapshot, best bid/ask price, and trading volume in the last 24 hours.
+func (by *Bybit) GetTickers(ctx context.Context, category, symbol, baseCoin string, expiryDate time.Time) (*TickerData, error) {
```

Comment:
This response does not capture the full API response. You need to check all responses:
<img width="1127" alt="image" src="https://github.com/thrasher-corp/gocryptotrader/assets/9261323/b9084fcc-0a04-43ed-8ce1-e6be8388c508">
The tabs there contain different pieces of data. Please include them.

If it gets too crazy for one struct, I'm happy if you split them up into their own API implementations if you'd rather